packages/ruleset/src/functions/unused-tag.js

```diff
@@ -0,0 +1,62 @@
+// Set of fields within a "path item" that we expect to hold an operation object.
+const operationMethods = [
+  'get',
+  'put',
+  'post',
+  'delete',
+  'options',
+  'head',
+  'patch',
+  'trace'
+];
+
+/**
+ * This function implements the 'unused-tag' validation rule.
+ * The specific checks that are performed are:
+ *
+ * 1. Each tag defined in the global "tags" field must be referenced by at least one.
+ *
+ * @param {object} rootDocument the entire API definition (assumed to be an OpenAPI 3 document)
+ * @returns an array of error objects
+ */
+function unusedTag(rootDocument) {
+  // Compile a list of all the tags.
+  const globalTags = rootDocument.tags || [];
```

Comment:
Should we bail here if `globalTags` is empty? Maybe an unnecessary optimization but thought I'd ask

Reply:
Yep, I'll add that check.